src/plone/restapi/deserializer/dxfields.py

```diff
@@ -45,9 +45,8 @@ class DatetimeFieldDeserializer(DefaultFieldDeserializer):
     def __call__(self, value):
         try:
             # Parse ISO 8601 string with Zope's DateTime module
-            # and convert to a timezone naive datetime in local time
-            value = DateTime(value).toZone(DateTime().localZone()).asdatetime(
-            ).replace(tzinfo=None)
+            # and convert to a UTC based timezone as datetime
+            value = DateTime(value).toZone('UTC').asdatetime()
```

Comment:
Again like above, I wouldn't use 'UTC' but the portal's default timezone.

Reply:
I think that the concept of having a "portal" timezone is wrong. In which case is p.a.event using it? It's ok if you always have users in the same timezone, but this is hardly the case in most user cases.

All datetimes should be referenced using a common baseline. Then the UI convert it to the user's timezone. If not, a user in India will see always the dates offsetted with the portal timezone, which is pretty useless for him. This is what happened before and that's the reason of this change.

Reply:
@thet I thought you agree on that too when you said: https://github.com/plone/plone.restapi/issues/253#issuecomment-324383766

Reply:
Yes, true. I was suggesting that because I thought you do not have a timezone information in the JSON object available - besides the offset.
But if you have timezone-naive datetimes, like in the code above - and you convert those to UTC, then you make a decision which could be wrong. Unless you always exchange dates in UTC.

Fact is, plone.app.event 2.x does not convert it's datetimes to UTC but stores them in the default timezone - which is:
1) The event's timezone (not implemented yet)
2) The user's timezone (probably most often not set)
3) The portal's timezone (You have to set this on portal creation time)

I agree that users from anywhere want to have the option to show the datetimes in their local zone. But that's a UI thing and should be done via JS with the option to show the datetimes in the original zone.

So: converting to UTC and storing in UTC is OK, if you have no timezone identifier (offsets are a bit useless) and exchange in UTC anyways.
IMO it's not OK if the datetimes are "naive" - without any timezone information.

Reply:
@sneridagh If you want, we can have a short hangout if you think I can help here...

Reply:
@thet sure, whenever you are available, we can have a talk about it. We should find a common agreement on this.

Agreed that the implementation makes an assumption that p.a.event is not (yet) doing but IMHO should do and we should fix it upstream in Plone. That is, save the internal dates with timezone information under a common frame of reference (say UTC).

Clients (and existing UI) should provide always datetimes depending on the user's timezone (we are already doing that in a client's project all Angular UI. Current libraries always transform to UTC on serializing and it's easy to be user's timezone aware.

Setting a portal default timezone (or even a user timezone) it's useless and error prone. The current implementation now it's broken, as you can see, the provided tests are not passing using the current version of plone.restapi. 

The more sensible solution (as agreed in #253 seems to store always the timezone information) but then, use a common reference for that (using server's timezone will also be error prone and dates could not be the same if you migrate one server from timezone).

@tisto @lukasgraf @buchi any insights on this?

Reply:
I would say, if the convention in plone.restapi is to always export dates as UTC that is an convention. If other modules for other export formats uses another convetion that might not be consistent, but possible.

I would prefer to always export datetime objects as timezone aware objects, and that transformed to UTC, but also to always import all native datetime input as new timezone aware datetime objects. Presentation in the UI should be bound to user settings.